### ISSUE TITLE
Add inert match for unicode

### ISSUE THREAD

**Issue by xBLACKICEx** [nushell/nushell#7009]
### Related problem

Nushell does not support matching Unicode by English (ASCII) characters
This is very annoying for CJK speakers in the terminal, who need to spend a lot of time to switch input to match.

Problem like:
![pasgfiojasdn](https://user-images.githubusercontent.com/51036094/200091944-8e639509-5490-499a-a3ab-ed8f87aa9028.gif)
The above example has been troublesome when I typed it, but the situation has been good, because the input method matches the first group of Chinese characters that I want. But in general it is not possible to match several combinations of "gong zuo" in Chinese.
![image](https://user-images.githubusercontent.com/51036094/200092583-8c86b046-125c-4b1a-b0cd-10be891fb029.png)

### Describe the solution you'd like

**The result I was hoping for:**

Inputting English letters(ASCII) should match the Unicode equivalent of the phonetic symbol or the inert match rule for that language.
微博(weibo)
![image](https://user-images.githubusercontent.com/51036094/200091998-9cf7293d-abe1-427a-acd8-2c83e305e1b8.png)

### Describe alternatives you've considered

I wrote a library in Rust that relies on the deunicode library to initially implement inert input support for Chinese pinyin, and to match other unicode phonetic alphabets.

![image](https://user-images.githubusercontent.com/51036094/200094434-86aac987-2f77-4102-8d36-025c92830777.png)

the repo:
https://github.com/xBLACKICEx/unicode_layzy_match

### Additional context and details

I tried to contact the carpace developers to try to get nushell support in a third-party complementary way. But he said it's more about when to replace the original ascii to unicode characters so it should be natively supported by nushell.
https://github.com/rsteube/carapace-spec

**Comment by fdncred** [comment]
My guess at this is that you'd want to make a config option that allows people to enable or disable this feature. I'm not exactly sure how the completion system works though.

**Comment by xBLACKICEx** [comment]
> My guess at this is that you'd want to make a config option (选项) that allows people to enable or disable (禁用) this feature (特征) . I'm not exactly sure how the completion (完成) system (系统) works (工作) though (虽然) .

Indeed, for Western European language speakers this feature is not necessary after all, the keyboard can be typed with one key.
But the options are an afterthought, I can't even add in nushll and do some basic testing right now. 
That's why I'm seeking help.😂

**Comment by rgwood** [comment]
I don't fully understand the problem, but we have some great Nushell developers who speak CJK languages.

@hustcer and @WindSoilder, do you have any ideas on how to implement this? Feel free to discuss in 中文 if that's easiest :)

**Comment by xBLACKICEx** [comment]
> I don't fully understand the problem, but we have some great Nushell developers who speak CJK languages.
> 
> @hustcer and @WindSoilder, do you have any ideas on how to implement this? Feel free to discuss in 中文 if that's easiest :)

In short:It is impossible to complete Unicode by letters!

A simple example: an English directory with [ Download, Games ]. If you want to enter the Download folder, you would usually type "D" + "Tab" to do so, but if you switch to Chinese, it won't work.

Chinese directory: [ 下载(xia zai), 游戏(you xi) ] If I type "x" + "tab" directly the autocomplete will not find anything, I have to type it all out.  so for 
![image](https://user-images.githubusercontent.com/51036094/206315341-16be4dae-5703-49cf-928f-bea72705da59.png)

Minimum input times from 2 times to 4 times if you count the switch input method minimum input at this point to 6 times (win + space)
